packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -2789,7 +2803,33 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
       onLiveTextInput: liveTextInputEnabled
           ? () => _startLiveTextInput(SelectionChangedCause.toolbar)
           : null,
-    );
+    )..addAll(_textProcessingActionButtonItems);
+  }
+
+  List<ContextMenuButtonItem> get _textProcessingActionButtonItems {
+    final List<ContextMenuButtonItem> buttonItems = <ContextMenuButtonItem>[];
+
+    if (widget.obscureText || textEditingValue.selection.isCollapsed) {
```

Comment:
Should we also not build the system wide buttons if the text selection is invalid?

Reply:
This check was inspired by the implementation of accessors such as `cutEnabled`, `copyEnabled`, etc. See

https://github.com/flutter/flutter/blob/cdc83e5ebae492b314cb06d6416162cc63a1dba6/packages/flutter/lib/src/widgets/editable_text.dart#L2244-L2249

Those accessors are called in `EditablTestState.contextMenuButtonItems` just before the call to `_textProcessingActionButtonItems` I added. Because this is meant to build the text selection toolbar, it would make sense that a check to `TextSelection.isValid` is done somewhere before.

I did not spot an obvious place where such a check is done so I have updated the PR to add this check on the line you commented.